packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  Address,
+  Customer,
+  PaymentSessionStatus,
+  ShippingMethod,
+} from "../models"
+import { MedusaContainer } from "../types/global"
+
+export type PaymentProcessorContext = {
+  shipping_address?: Address | null
+  shipping_methods?: ShippingMethod[]
+  billing_address?: Address | null
+  email: string
+  currency_code: string
+  amount: number
+  resource_id?: string
+  customer?: Customer
+}
+
+export type PaymentProcessorSessionResponse = {
+  update_requests: { customer_metadata: Record<string, unknown> }
+  session_data: Record<string, unknown>
+}
+
+export interface PaymentProcessorError {
+  error: string
+  code: number
+  details: any
+}
+
+/**
+ * The new payment service plugin interface
+ */
+export interface PaymentProcessor {
+  /**
+   * Return a unique identifier to retrieve the payment plugin provider
+   */
+  getIdentifier(): string
```

Comment:
thought: A scenario came up the other day where a user needed to register two of the same plugins to their store. Specifically, they had two PayPal accounts: 1 for EUR and 1 for USD. It would be valuable if we could open up for customizing the identifier through `medusa-config.js`'s plugin options. 

This probably has more to do with how we register plugins.

Reply:
yes got it,

Reply:
I would have assumed that this is something managed by the Paypal API depending on the currency that is used. Don't you think it would need a bit more investigation?

Reply:
I think @srindom's idea is excellent, but I suggest we keep it for a separate ticket to not block this PR further, as we would prob. need to rethink payment plugin registration.